LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+			LocalyticsClass.CallStatic("setOption", key, new AndroidJavaObject("java/lang/Boolean", boolValue));
+			#elif UNITY_IOS
+			_setBoolOption(key, boolValue);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		public static string AppKey
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<string>("getAppKey");
+				#elif UNITY_IOS
+				return _appKey();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static string InstallId
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<string>("getInstallId");
+				#elif UNITY_IOS
+				return _installId();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static string LibraryVersion
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<string>("getLibraryVersion");
+				#elif UNITY_IOS
+				return _libraryVersion();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static bool LoggingEnabled
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<bool>("isLoggingEnabled");
+				#elif UNITY_IOS
+				return _isLoggingEnabled();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+			set
+			{
+				#if UNITY_ANDROID
+				LocalyticsClass.CallStatic("setLoggingEnabled", value);
+				#elif UNITY_IOS
+				_setLoggingEnabled(value);
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+#if UNITY_ANDROID
+		
+		/*
+		 * Android Helpers
+		 *
+		 **/
+		private static AnalyticsListener _analyticsListener;
+		private static MessagingListener _messagingListener;
+		private static LocationListener _locationListener;
+
+		private static AndroidJavaClass _localyticsClass;
+		private static AndroidJavaClass LocalyticsClass
+		{
+			get
+			{
+				if (_localyticsClass == null)
+				{
+					_localyticsClass = new AndroidJavaClass("com.localytics.android.Localytics");
+				}
+				return _localyticsClass;
+			}
+		}
+
+		private static AndroidJavaObject DictionaryToMap(Dictionary<string, string> dict)
+		{
+			var map = new AndroidJavaObject("java.util.HashMap");
+
+			if (dict == null || dict.Count == 0)
+			{
+				return map;
+			}
+
+			IntPtr putMethod = AndroidJNIHelper.GetMethodID(map.GetRawClass(), "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+			var args = new object[2];
+			foreach (KeyValuePair<string, string> kvp in dict)
+			{
+				using (var k = new AndroidJavaObject("java.lang.String", kvp.Key))
+				{
+					using (var v = new AndroidJavaObject("java.lang.String", kvp.Value))
```

Comment:
This was the style that was used on some of these conversion methods before I came into this code, so I just continued using it.

Either way, it has nothing to do with garbage collection. In C# this is called [using alias directives](https://msdn.microsoft.com/en-us/library/aa664765(v=vs.71).aspx). Just allows us to use `v` instead of `kvp.Value` for cleaner code later.